### DISCUSSION TITLE
How to extract heading names from PDF file ?

### DISCUSSION THREAD

**Discussion by Laxmi530** [pikepdf/pikepdf#459] [Q&A]
Hai,
Thank you for providing a beautiful library. I am new to pikepdf and this library fells me bit complex. 
I am trying to extract the heading names from the pdf file and gone through the documentation but still did not get anything so, can someone please guide me is there any method or class available to do this task. 
Below attached the sample pdf file.

Thank you in advance.

[Sample PDF file.pdf](https://github.com/pikepdf/pikepdf/files/11001873/Sample.PDF.file.pdf)

**Comment by jbarlow83** [comment]
Unfortunately extracted structured information from a PDF is quite complex. PDF only describes the final appearance of a document, and has no concept of what text is doing. Even deciding where word breaks are is not straightforward.

You can use a library like pdfminer.six to extract text, and from here you may be able to use some heuristic such as checking for font that is larger than average, and classify that as a header. 

You could also a commercial PDF editor to create a Tagged PDF, in which you label all PDF text as belonging to a class similar to HTML.

pikepdf is a library that could be used for this sort of work, but it's domain specific and complex work that needs to be fine tuned for incoming PDFs.

**Reply by Laxmi530** [reply]
@jbarlow83 Thanks for the response.
Can you please guide me how to get the font name and font size using pikepdf.

Thank you

**Reply by jbarlow83** [reply]
I recommend using pdfminer.six instead for that task.

pikepdf is a different tool, in particular given its ability to modify PDFs in standards compliant ways. pdfminer.six cannot output PDFs at all, but it is great at text extraction.

**Comment by lpi-tn** [comment]
You can use pikepdf or any pdf parser + refinedoc, I built it especially for this case !
https://github.com/CyberCRI/refinedoc
